solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -122,6 +148,7 @@ pub type InnerClient = String; // Serialized
 pub type InnerConnectionEnd = String; // Serialized
 pub type InnerChannelEnd = String; // Serialized
 pub type InnerConsensusState = String; // Serialized
+pub type InnerPacketMsg = Vec<u8>;
```

Comment:
why stored it serialized?

Reply:
Because the `PacketMsg` doesnt implement `AnchorSerialize` and `AnchorDeserialize` macros

Reply:
[AnchorSerialize and AnchorDeserialize are just aliases to Borsh](https://github.com/coral-xyz/anchor/blob/7e5bdce710f9580e1489089d04afb13f71a5d1c7/lang/src/lib.rs#L58) so PacketMsg does actually implement it.